app/controllers/api/v1/accounts_controller.rb

```diff
@@ -1,7 +1,8 @@
 # frozen_string_literal: true
 
 class Api::V1::AccountsController < ApiController
-  before_action -> { doorkeeper_authorize! :read }, except: [:follow, :unfollow, :block, :unblock, :mute, :unmute, :update_credentials]
+  before_action -> { doorkeeper_authorize_if_got_token! :read }, only: [:show, :following, :followers, :statuses, :search]
+  before_action -> { doorkeeper_authorize! :read }, except: [:show, :following, :followers, :statuses, :search, :follow, :unfollow, :block, :unblock, :mute, :unmute, :update_credentials]
```

Comment:
If you've got that many things in the `except:` clause, isn't it easier to use `only:` instead?